sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -418,6 +426,19 @@ class JacksonParser(
       }
     }
 
+    // When the input schema is setting to `nullable = false`, make sure the field is not null.
+    var index = 0
+    while (badRecordException.isEmpty && !skipRow && index < schema.length) {
+      if (!schema(index).nullable && row.isNullAt(index)) {
+        throw new IllegalSchemaArgumentException(
+          s"the null value found when parsing non-nullable field ${schema(index).name}.")
```

Comment:
nit: `field ${schema(index).name} is not nullable but it's missing in one record.`

Reply:
done.